modules/pxp-module-puppet

```diff
@@ -70,7 +70,8 @@ end
 
 def force_unicode(s)
   # Puppet still returns UTF-8 when locale is C or POSIX, so force encoding to that.
-  if s.encoding == Encoding::ASCII_8BIT
+  if !Puppet.features.microsoft_windows? &&
+    (s.encoding == Encoding::US_ASCII || s.encoding == Encoding::ASCII_8BIT)
```

Comment:
pxp-agent (and thus Puppet) is operating in a POSIX locale as the default on some systems (SLES 10/11, arista, cumulus vx), with default_external encoding as US-ASCII. It prints UTF-8 because that's present in the manifest. Either we consider this a non-issue and change the testing, or we have to force_encoding to UTF-8. I think the only question here is how narrowly we target the change.

`if (s.encoding == Encoding::US_ASCII)` is sufficient to address all the problem platforms we test, and should be perfectly safe (as UTF-8 is a superset of US-ASCII).